core/src/main/java/org/apache/iceberg/rest/RestClient.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.io.Closeable;
+
+/**
+ * Eventual Interface for pluggable RestClients.
```

Comment:
Do you mean to allow customizing the HTTP client implementation?

Reply:
Yes. That's what I mean. People might have a preference on the HTTP client implementation. Particularly if they use HTTP2 or HTTP3 even. So long as it conforms to the spec, it should in theory be fine.

But that was my thinking.

Reply:
I think we can actually learn a bit from the HTTP client in AWS SDK v2, it supports plugging in any HTTP client implementation. We can borrow the same design pattern here.